File: python/credit_check.py
def credit_check(s):
    total = 0
    for index, number in enumerate(s):
        num = int(number)
        if index%2 == 0:
            num *= 2
        if num >= 10:
            temp = 0
            for char in str(num):
                temp += int(char)
            num = temp
        total += num
    if  total % 10 == 0:
        return "The number is valid!"
    else:
        return "The number is invalid!"

File: js/creditCheck.js
exports.creditCheck = function(str) {
  let total = 0;
  str.split("").forEach((s,index) => {
    let num = parseInt(s);
    if (index % 2 == 0){
      num *= 2;
    }
    if (num >= 10) {
      let digits = num.toString().split("");
      num = digits.reduce((acc, current) => {
        return acc + parseInt(current);
      }, 0)
    }
    total += num;
  })
  
  let remain = total % 10;
  if (remain == 0){
    return "The number is valid!"
  } else {
    return "The number is invalid!"
  }
}